bitcoin/examples/sighash.rs

```diff
@@ -0,0 +1,199 @@
+use bitcoin::{consensus, ecdsa, sighash, Amount, PublicKey, Script, ScriptBuf, Transaction};
+use hex_lit::hex;
+
+//These are real blockchain transactions examples of computing sighash for:
+// - P2WPKH
+// - P2MS 2of3
+// - P2SH 2of2 multisig
+// - P2WSH 2of2 multisig
+
+//run with: cargo run --example sighash
+
+//TODO add P2TR examples, ideally for both key-path and script-path spending
+
+/// Computes segwit sighash for a transaction input that spends a p2wpkh output with "witness_v0_keyhash" scriptPubKey.type
+///
+/// # Arguments
+///
+/// * `raw_tx` - spending tx hex
+/// * `inp_idx` - spending tx input index
+/// * `value` - ref tx output value in sats
+fn compute_sighash_p2wpkh(raw_tx: &[u8], inp_idx: usize, value: u64) {
+    let tx: Transaction = consensus::deserialize(raw_tx).unwrap();
+    let inp = &tx.input[inp_idx];
+    let witness = &inp.witness;
+    println!("Witness: {:?}", witness);
```

Comment:
Printing in hex is perhaps nicer. One way to achieve that is consensus serializing then using the `as_hex` function.
```suggestion
    println!("Witness: {:x}", consensus::serialize(&witness).as_hex());
```

And you'll need to: `use internals::hex::display::DisplayHex;`

Reply:
This comment is just an FYI because I did it while playing with https://github.com/rust-bitcoin/rust-bitcoin/pull/1913

Reply:
I think it's just fine to wait for #1913.

Reply:
> Printing in hex is perhaps nicer. One way to achieve that is consensus serializing then using the `as_hex` function.
> 
> And you'll need to: `use internals::hex::display::DisplayHex;`

That's better, for sure. 

Will you merge this now because it has two acks (before https://github.com/rust-bitcoin/rust-bitcoin/pull/1913 ) , or should I make this change (I will be away till Monday)?